mesonbuild/compilers/compilers.py

```diff
@@ -1793,7 +1794,7 @@ def openmp_flags(self):
 
 class ArmclangCompiler:
     def __init__(self, compiler_type):
-        if not self.is_cross:
+        if not self.for_machine >= MachineChoice.HOST:
```

Comment:
This is so complex with `not >=`, couldn't this just be `if self.for_machine is not MachineChoice.BUILD`?

Reply:
I'm just going to set this back to `is_cross`. `Compiler` really shouldn't have `for_machine`, but just `machine`. In other words, the compiler targets some machine, period, and comes prior to what our build, host, and target, machines are.

Then of course, the assert here becomes `compiler.machine.is_arm()` rather than some `is_cross` nonsense.